src/main/java/uk/gov/gchq/palisade/service/request/Policy.java

```diff
@@ -0,0 +1,229 @@
+
+/**
+ * This class is used to store the information that is required by the policy
+ * service but not needed by the rest of the palisade services. That includes
+ * separating the rules that need to be applied at the resource level or the record level.
+ *
+ * @param <T> The Java class that the rules expect the records of data to be in
+ *            the format of, e.g. if T was String then a policy has Rules of type Resource for coarse filtering
+ *            and Rules of type String for fine grain filtering of files where each record is a String
+ */
+public class Policy<T> {
+    private Rules<T> recordRules;
+    private Rules<Resource> resourceRules;
+    private User owner;
+
+    // no-args constructor required
+    public Policy() {
+        recordRules = new Rules<>();
+        resourceRules = new Rules<>();
+    }
+
+    public Policy<T> recordRules(final Rules<T> recordRules) {
+        requireNonNull(recordRules, "The record level rules cannot be set to null.");
+        this.recordRules = recordRules;
+        return this;
+    }
+
+    public Policy<T> resourceRules(final Rules<Resource> resourceRules) {
+        requireNonNull(resourceRules, "The resource level rules cannot be set to null.");
+        this.resourceRules = resourceRules;
+        return this;
+    }
+
+    @JsonIgnore
+    public String getMessage() {
+        return "Resource level rules: " + getResourceRules().getMessage() + ", record level rules: " + getRecordRules().getMessage();
+    }
+
+    public Rules<T> getRecordRules() {
+        // will never be null
+        return recordRules;
+    }
+
+    public void setRecordRules(final Rules<T> recordRules) {
+        recordRules(recordRules);
+    }
+
+    public Rules<Resource> getResourceRules() {
+        // will never be null
+        return resourceRules;
+    }
+
+    public void setResourceRules(final Rules<Resource> resourceRules) {
+        resourceRules(resourceRules);
+    }
+
+    private static String generateUUID() {
+        return UUID.randomUUID().toString();
+    }
+
+    private void addMessage(final String newMessage, final Rules rules) {
+        requireNonNull(newMessage, "Cannot add a null message.");
+        requireNonNull(rules, "Cannot add a message to a null set of rules.");
+        String currentMessage = rules.getMessage();
+        if (currentMessage.equals(Rules.NO_RULES_SET)) {
+            rules.message(newMessage);
+        } else {
+            rules.message(currentMessage + ", " + newMessage);
+        }
+    }
+
+    public Policy<T> recordLevelRule(final String message, final Rule<T> rule) {
+        requireNonNull(message, "The message cannot be null and should indicate what the rule is doing.");
+        requireNonNull(rule, "Cannot set a null rule.");
+        Rules<T> recordLevelRules = getRecordRules();
+        recordLevelRules.rule(generateUUID(), rule);
+        addMessage(message, recordLevelRules);
+        return this;
+    }
```

Comment:
Presumably a lot of these constructor-like methods are not needed, can they be removed?